cmd/stateless/main.go

```diff
@@ -0,0 +1,234 @@
+package main
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/console/prompt"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/eth/ethconfig"
+	"github.com/ethereum/go-ethereum/internal/debug"
+	"github.com/ethereum/go-ethereum/internal/flags"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/trie"
+	"github.com/urfave/cli/v2"
+	"go.uber.org/automaxprocs/maxprocs"
+	"os"
+)
+
+var (
+	BlockWitnessFlag = &cli.StringFlag{
+		Name:  "block-witness",
+		Usage: "foo bar",
+	}
+
+	BlockWitness1Flag = &cli.StringFlag{
+		Name:  "witness1",
+		Usage: "foo bar",
+	}
+	BlockWitness2Flag = &cli.StringFlag{
+		Name:  "witness2",
+		Usage: "foo bar",
+	}
+
+	WitnessDiffCommand = &cli.Command{
+		Action:    witnessDiff,
+		Name:      "diff",
+		Usage:     "",
+		ArgsUsage: "<genesisPath>",
+		Flags: []cli.Flag{
+			BlockWitness1Flag,
+			BlockWitness2Flag,
+		},
+		Description: `placeholder description`,
+	}
+	PPCommand = &cli.Command{
+		Action:    pp,
+		Name:      "pp",
+		Usage:     "",
+		ArgsUsage: "<genesisPath>",
+		Flags: []cli.Flag{
+			BlockWitnessFlag,
+		},
+		Description: `placeholder description`,
+	}
+	StatelessCommand = &cli.Command{
+		Action:    statelessCmd,
+		Name:      "exec",
+		Usage:     "",
+		ArgsUsage: "<genesisPath>",
+		Flags: []cli.Flag{
+			BlockWitnessFlag,
+		},
+		Description: `placeholder description`,
+	}
+	ServerCommand = &cli.Command{
+		Action:      server,
+		Name:        "server",
+		Usage:       "",
+		ArgsUsage:   "<genesisPath>",
+		Flags:       []cli.Flag{},
+		Description: `placeholder description`,
+	}
+)
+
+var app = flags.NewApp("stateless block executor")
+
+func init() {
+	// Initialize the CLI app and start Geth
+	app.Copyright = "Copyright 2013-2023 The go-ethereum Authors"
+	app.Commands = []*cli.Command{
+		WitnessDiffCommand,
+		PPCommand,
+		StatelessCommand,
+		ServerCommand,
+	}
+
+	app.Flags = []cli.Flag{
+		BlockWitnessFlag,
+	}
+
+	app.Before = func(ctx *cli.Context) error {
+		maxprocs.Set() // Automatically set GOMAXPROCS to match Linux container CPU quota.
+		if err := debug.Setup(ctx); err != nil {
+			return err
+		}
+		return nil
+	}
+	app.After = func(ctx *cli.Context) error {
+		debug.Exit()
+		prompt.Stdin.Close() // Resets terminal mode.
+		return nil
+	}
+}
+
+func statelessCmd(ctx *cli.Context) error {
+	blockWitnessPath := ctx.String(BlockWitnessFlag.Name)
+	if blockWitnessPath == "" {
+		panic("block witness required")
```

Comment:
Isn't exit with `1` sufficient? You can just return the error and it will be taken care of. Having a stacktrace at this point is moot, we know what the error is and what caused it